components/tag-list/tag-list.js

```diff
@@ -406,7 +410,22 @@ class TagList extends LocalizeCoreElement(InteractiveMixin(ArrowKeysMixin(LitEle
 		this._items[0].setAttribute('keyboard-tooltip-item', true);
 		if (this._hasShownKeyboardTooltip) this._items[0].setAttribute('keyboard-tooltip-shown', true);
 		await this.updateComplete;
-		refocus.focus?.();
+		if (this._refocus?.classList.contains('d2l-tag-list-button')) {
+			this._refocus = this.shadowRoot.querySelector('.d2l-tag-list-button');
+		}
+		await this._refocus?.updateComplete;
+		(this._refocus || refocus).focus?.();
+		this._refocus = null;
+	}
+
+	_handleSlotFocusIn() {
+		this._items[0].setAttribute('tabindex', '-1');
```

Comment:
If the `slotchange` event runs while focus is inside the slot, this block of code (above) would undo this assignment. It looks like the slot's `focusin`/`focusout` handlers run when an item is removed via keyboard. As far as I know, there wouldn't be any other way for the `slotchange` event to be triggered while focus is already inside the slot, so maybe we're covered. It's not like a tag would be added while focus is inside... that would require the user to something with focus outside the slot.

```javascript
this._items.forEach((item, index) => {
	item.setAttribute('tabIndex', index === 0 ? 0 : -1);
});
```

Reply:
I think we do want this to run in that case, so if you press `Shift+Tab` from the middle of the list you don't go to the first item before exiting the list. Maybe I'm wrong about that though?